### PR TITLE
Logging logic for user interactions with the agent

### PR DESCRIPTION
**Pull request by jhan6501**
# Description
Add logging logic for logging interactions with the agent. 
Interactions get saved as a json in interaction_loggings.json

Interactions get saved by reading the interaction_loggings.json file, storing the json in a variable, editing that variable, then redumping the new variable in the old file. This is inefficient (due to the reading). I want to eventually get to the point where we do not read the file, and can just append directly without reading (or could save the data to a mongo database).

Fixes # (issue)

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After
Should not change existing features, purely new
image

# Testing
Started server on 8000 and 3000, on 3000 sent commands to agent with the interact app, and annotated some objects, then checked the interaction_loggings.json file

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.

### CODE REVIEW

File: agents/locobot/locobot_agent.py [hunk 1]
```diff
@@ -122,7 +122,7 @@ def objects_in_memory(sid):
                 del o["feature_repr"] # pickling optimization
             self.dashboard_memory["objects"] = objects
             sio.emit("updateState", {"memory": self.dashboard_memory})
-
+
```
Comment (Rebecca-Qian) [inline agents/locobot/locobot_agent.py:125]: nit: why a newline here?

File: droidlet/dashboard/web/src/StateManager.js [hunk 2]
```diff
@@ -184,11 +197,11 @@ class StateManager {
     socket.on("image", this.processRGBDepth); // RGB + Depth
 
     socket.on("objects", this.processObjects);
-    
+
     socket.on("updateVoxelWorldState", this.updateVoxelWorld);
     socket.on("setVoxelWorldInitialState", this.setVoxelWorldInitialState);
     socket.on("showAssistantReply", this.showAssistantReply);
-    
+
```
Comment (Rebecca-Qian) [inline droidlet/dashboard/web/src/StateManager.js:204]: nit: why the newlines?

File: droidlet/dashboard/web/src/StateManager.js [hunk 3]
```diff
@@ -352,6 +365,18 @@ class StateManager {
     }
   }
 
+  /**
+   * data is a dictionary
```
Comment (Rebecca-Qian) [inline droidlet/dashboard/web/src/StateManager.js:369]: Can you make this function header more descriptive?

File: droidlet/dashboard/web/src/components/Interact/Message.js [hunk 4]
```diff
@@ -119,6 +119,8 @@ class Message extends Component {
     if (chatmsg.replace(/\s/g, "") !== "") {
       //add to chat history box of parent
       this.props.setInteractState({ msg: chatmsg, failed: false });
+      //log message to flask
+      this.props.stateManager.logInteractiondata("text commad", chatmsg);
```
Comment (Rebecca-Qian) [inline droidlet/dashboard/web/src/components/Interact/Message.js:123]: Typo

File: droidlet/perception/semantic_parsing/utils/interaction_logger.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
+"""
+Copyright (c) Facebook, Inc. and its affiliates.
+"""
+
+import json
+
+class interactionLogger:
```
Comment (Rebecca-Qian) [inline droidlet/perception/semantic_parsing/utils/interaction_logger.py:7]: capital for Interaction

File: .gitignore [hunk 6]
```diff
@@ -44,6 +44,7 @@ incoming_chat.txt
 locobot/test/test_assets/perception_handlers/
 locobot/agent/dashboard_data.db*
 nsp_outputs.csv
+agents/locobot/interaction_loggings.json
```
Comment (Rebecca-Qian) [inline .gitignore:47]: Are there more file outputs that you don't want to commit? This can match `*.json` for example

Comment (jhan6501) [reply .gitignore:47]: I do not think so, since this is the only new file I introduced to the codebase

**Review by Rebecca-Qian — APPROVED** [review]
LGTM!

**Pull request merged**